src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -820,6 +828,72 @@ def update_asg(instance, tags=None):
 # endregion
 
 
+# region DdosProtectionPlans
+def create_ddos_plan(cmd, resource_group_name, ddos_plan_name, location=None, tags=None, vnets=None):
+    from azure.cli.core.commands import LongRunningOperation
+
+    ddos_client = network_client_factory(cmd.cli_ctx).ddos_protection_plans
+    DdosProtectionPlan, SubResource = cmd.get_models('DdosProtectionPlan', 'SubResource')
+    plan = DdosProtectionPlan(
+        name=ddos_plan_name,
+        location=location,
+        tags=tags
+    )
+    if not vnets:
+        # if no VNETs can do a simple PUT
+        return ddos_client.create_or_update(resource_group_name, ddos_plan_name, plan)
+
+    # if VNETs specified, have to create the protection plan and then add the VNETs
+    plan_id = LongRunningOperation(cmd.cli_ctx)(
+        ddos_client.create_or_update(resource_group_name, ddos_plan_name, plan)).id
+
+    logger.info('Attempting to attach VNets to newly created DDoS protection plan.')
+    for subresource in vnets:
+        vnet_client = network_client_factory(cmd.cli_ctx).virtual_networks
+        id_parts = parse_resource_id(subresource.id)
+        vnet = vnet_client.get(id_parts['resource_group'], id_parts['name'])
+        vnet.ddos_protection_plan = SubResource(id=plan_id)
+        vnet_client.create_or_update(id_parts['resource_group'], id_parts['name'], vnet)
+    return ddos_client.get(resource_group_name, ddos_plan_name)
+
+
+def update_ddos_plan(cmd, instance, tags=None, vnets=None):
+    SubResource = cmd.get_models('SubResource')
+
+    if tags is not None:
+        instance.tags = tags
+    if vnets == "":
+        vnets = []
+    if vnets is not None:
+        logger.info('Attempting to update the VNets attached to the DDoS protection plan.')
+        vnet_ids = [x.id for x in vnets]
```

Comment:
Please double check this, as I have been a bit lost for what the type of `vnets` argument really is. If it is a string array, then the code will crash

Reply:
It's a list of SubResources, because the existing VNET validator does that.